proof/dleq_test.go

```diff
@@ -0,0 +1,106 @@
+package proof_test
```

Comment:
If this is to be used widely, I'd add some test that also checks for bad proofs.